pkg/operation/botanist/kubeapiserverexposure.go

```diff
@@ -151,3 +151,32 @@ func (b *Botanist) setAPIServerServiceClusterIP(clusterIP string) {
 		},
 	)
 }
+
+// DeployKubeAPIServerIngress deploys ingress for the kube-apiserver.
+func (b *Botanist) DeployKubeAPIServerIngress(ctx context.Context) error {
+	// Do not deploy ingress if there is no wildcard certificate
+	if b.ControlPlaneWildcardCert == nil {
+		return b.DestroyKubeAPIServerIngress(ctx)
+	}
+	ingressClass, err := gardenerutils.ComputeNginxIngressClassForSeed(b.Seed.GetInfo(), b.Seed.GetInfo().Status.KubernetesVersion)
+	if err != nil {
+		return err
+	}
+	return kubeapiserverexposure.NewIngress(
```

Comment:
Instead of creating the component on the fly in these functions, can you add it to https://github.com/gardener/gardener/blob/d21b3ccfa1b6d76e483ba83b6f292d232ca35e55/pkg/operation/shoot/types.go#L117-L132 and create it [here](https://github.com/gardener/gardener/blob/d21b3ccfa1b6d76e483ba83b6f292d232ca35e55/pkg/operation/botanist/botanist.go#L133-L135)?